### PR TITLE
Upgrade azure-mgmt-resources to v2 runtime

### PR DESCRIPTION
**Pull request by RikkiGibson**
Continuation of #127, containing changes to manually-written code. Handling gradual migration by commenting out child modules which have not been converted yet.

### CODE REVIEW

**Comment by RikkiGibson** [comment]
@jianghaolu do you know how I can test ProviderRegistrationInterceptor now that I've rewritten it for v2?

**Comment by jianghaolu** [comment]
You can unregister a provider temporarily in Azure CLI through `az provider unregister`.

**Comment by RikkiGibson** [comment]
Remaining failures are due to unnecessary imports in Inners. Will track them down in the generator.

**Comment by anuchandy** [comment]
@RikkiGibson sorry for the delay, I am looking into this PR.

File: azure-mgmt-resources/src/main/java/com/microsoft/azure/v2/management/resources/fluentcore/arm/models/implementation/ExternalChildResourceImpl.java [hunk 1]
```diff
@@ -102,7 +103,7 @@ public PendingOperation pendingOperation() {
      *
      * @param pendingOperation the new state of this child resource
      */
-    public void setPendingOperation(PendingOperation pendingOperation) {
+    public synchronized void setPendingOperation(PendingOperation pendingOperation) {
         this.pendingOperation = pendingOperation;
```
[the remark this answers is not in the document]
Comment (RikkiGibson) [reply azure-mgmt-resources/src/main/java/com/microsoft/azure/v2/management/resources/fluentcore/arm/models/implementation/ExternalChildResourceImpl.java]: This was added a little haphazardly, but the thought behind it is that we need to be very careful about any mutation of shared state in the mapper we pass to flatMap. The Java memory model requires that you use some kind of synchronization if you need thread B to see the most recent value written to a location by thread A. Without it we may have a program that works 99.99% but will cause transient failures precisely under the high-load production conditions where it's hard to troubleshoot such failures.

File: azure-mgmt-resources/src/main/java/com/microsoft/azure/v2/management/resources/fluentcore/model/Executable.java [hunk 2]
```diff
@@ -33,7 +33,7 @@
      * @return the handle to the REST call
      */
     @Method
-    Observable<T> executeAsync();
+    Single<T> executeAsync();
```
[the remark this answers is not in the document]
Comment (RikkiGibson) [reply azure-mgmt-resources/src/main/java/com/microsoft/azure/v2/management/resources/fluentcore/model/Executable.java]: Thank you. I overlooked this one.

**Pull request closed without merging**